ibis/backends/pyarrow/datatypes.py

```diff
@@ -43,6 +42,29 @@ def to_pyarrow_type(dtype: dt.DataType) -> pa.DataType:
     return arrow_type
 
 
+_IBIS_TO_PYARROW_UNIT = {
+    None: "ms",
+    0: "s",
+    1: "ms",
+    2: "ms",
+    3: "ms",
+    4: "us",
+    5: "us",
+    6: "us",
+    7: "ns",
+    8: "ns",
+    9: "ns",
+}
+
+
+@to_pyarrow_type.register(dt.Timestamp)
+def from_ibis_timestamp(dtype: dt.Timestamp) -> pa.TimestampType:
```

Comment:
This change will conflict with the `formats` PRs. I've added this functionality at https://github.com/ibis-project/ibis/pull/6202/files#diff-4d8d4c2cba315d883d9dea888bb2c49d2756b1fd225d5f1600a8c490ef1851adR463

Reply:
I suppose I can xfail whatever fails here, and then it will start to `XPASS` in your PR.